contracts/Staking.sol

```diff
@@ -113,148 +97,181 @@ contract Staking is Autopetrified, ERCStaking, ERCStakingHistory, IStakingLockin
      * @param _accountAddress Owner of locked tokens
-     * @param _lockId Id of the lock for the given account
-     * @param _newAmount New amount of locked tokens
+     * @param _lockManager The manager entity for this particular lock
+     * @param _allowance Amount of allowed tokens decrease
      */
-    function decreaseLockAmount(address _accountAddress, uint256 _lockId, uint256 _newAmount) external isLockManager(_accountAddress, _lockId) {
-        // lock 0 tokens makes no sense
-        require(_newAmount > 0, ERROR_AMOUNT_ZERO);
+    function decreaseLockAllowance(address _accountAddress, address _lockManager, uint256 _allowance) external isInitialized {
+        // only owner and manager can decrease allowance
+        require(msg.sender == _accountAddress || msg.sender == _lockManager, ERROR_CANNOT_CHANGE_ALLOWANCE);
+        require(_allowance > 0, ERROR_AMOUNT_ZERO);
 
-        // manager can only decrease locked amount
-        Lock storage lock_ = accounts[_accountAddress].locks[_lockId];
-        require(_newAmount < lock_.amount, ERROR_INCREASING_LOCK_AMOUNT);
+        Lock storage lock_ = accounts[_accountAddress].locks[_lockManager];
+        uint256 newAllowance = lock_.allowance.sub(_allowance);
+        require(newAllowance >= lock_.amount, ERROR_NOT_ENOUGH_ALLOWANCE);
+        // unlock must be used for this:
+        require(newAllowance > 0, ERROR_ALLOWANCE_ZERO);
 
-        lock_.amount = _newAmount;
-        emit LockAmountChanged(_accountAddress, _lockId, _newAmount);
+        lock_.allowance = newAllowance;
+
+        emit LockAllowanceChanged(_accountAddress, _lockManager, _allowance, false);
     }
 
     /**
-     * @notice Change the manager of `_accountAddress`'s lock #`_lockId` to `_newManager`
+     * @notice Increase `_amount` tokens from lock of `_accountAddress` by `_lockManager`
+     * @param _accountAddress Owner of locked tokens
+     * @param _lockManager The manager entity for this particular lock
+     * @param _amount Amount of locked tokens increase
+     */
+    function increaseLockAmount(address _accountAddress, address _lockManager, uint256 _amount) external isInitialized {
+        require(_amount > 0, ERROR_AMOUNT_ZERO);
+
+        // check enough unlocked tokens are available
+        require(_amount <= _unlockedBalanceOf(_accountAddress), ERROR_NOT_ENOUGH_BALANCE);
+
+        _increaseActiveLockAmount(msg.sender, _lockManager, _amount);
+
+        emit LockAmountChanged(msg.sender, _lockManager, _amount, true);
+    }
+
+    /**
+     * @notice Decrease `_amount` tokens from lock of `_accountAddress` by `_lockManager`
+     * @param _accountAddress Owner of locked tokens
+     * @param _lockManager The manager entity for this particular lock
+     * @param _amount Amount of locked tokens decrease
+     */
+    function decreaseLockAmount(address _accountAddress, address _lockManager, uint256 _amount) external isInitialized {
```

Comment:
same here, I expect these to be called from the lock manager only

Reply:
Why “only”? If the tokens can be unlocked (imagine a time lock), the owner should be able to do it too.